apps/judicial-system/web/src/routes/Prosecutor/RestrictionRequest/StepTwo/StepTwo.tsx

```diff
@@ -83,30 +87,38 @@ export const StepTwo: React.FC = () => {
     })
 
   const handleNextButtonClick = async () => {
-    if (!workingCase) {
-      return
-    }
+    try {
+      if (!workingCase) {
+        return
+      }
 
-    const caseOpened =
-      workingCase.state === CaseState.NEW
-        ? await transitionCase(workingCase, CaseTransition.OPEN, setWorkingCase)
-        : true
-
-    if (caseOpened) {
-      if (
-        (workingCase.state !== CaseState.NEW &&
-          workingCase.state !== CaseState.DRAFT) ||
-        // TODO: Ignore failed notifications
-        workingCase.notifications?.find(
-          (notification) => notification.type === NotificationType.HEADS_UP,
-        )
-      ) {
-        router.push(`${Constants.STEP_THREE_ROUTE}/${workingCase.id}`)
+      const caseOpened =
+        workingCase.state === CaseState.NEW
+          ? await transitionCase(
+              workingCase,
+              CaseTransition.OPEN,
+              setWorkingCase,
+            )
+          : true
+
+      if (caseOpened) {
+        if (
+          (workingCase.state !== CaseState.NEW &&
+            workingCase.state !== CaseState.DRAFT) ||
+          // TODO: Ignore failed notifications
+          workingCase.notifications?.find(
+            (notification) => notification.type === NotificationType.HEADS_UP,
+          )
+        ) {
+          router.push(`${Constants.STEP_THREE_ROUTE}/${workingCase.id}`)
+        } else {
+          setModalVisible(true)
+        }
       } else {
-        setModalVisible(true)
+        // TODO: Handle error
       }
-    } else {
-      // TODO: Handle error
+    } catch (error) {
```

Comment:
Consider removing this try / catch as the error is already handled in the useCase hook.